File: src/main/java/com/cesello/observer/MessageStream.java
package com.cesello.observer;

import java.util.Deque;
import java.util.LinkedList;

public class MessageStream extends Subject {

    private Deque<String> messageHistory = new LinkedList<>();

    @Override
    void setState(String message) {
        messageHistory.add(message);
        this.notifyObservers();
    }

    @Override
    String getState() {
        return messageHistory.getLast();
    }
}

File: src/main/kotlin/com/cesello/observer/MessageStream.kt
package com.cesello.observer

import java.util.Deque
import java.util.LinkedList

class MessageStream(private val messageHistory: Deque<String> = LinkedList()): Subject() {
    override fun setState(message: String) {
        messageHistory.add(message)
        this.notifyObservers()
    }

    override fun getState(): String = messageHistory.last

}